apps/web/src/lib/monitoring/collectors/base.ts

```
@@ -0,0 +1,280 @@
+  protected emitMetric(
+    name: string,
+    value: number,
+    unit: string,
+    tags: Record<string, string | number> = {},
+    metadata?: Record<string, unknown>
+  ): void {
+    // Validate inputs
+    if (typeof name !== 'string' || name.length === 0) {
+      throw new Error('Metric name must be a non-empty string');
+    }
+    if (typeof value !== 'number' || !isFinite(value)) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)